services/gundeck/test/integration/API.hs

```diff
@@ -801,16 +857,30 @@ buildPush sdr rcps pload =
   where
     rcpt u c = recipient u RouteAny & recipientClients .~ c
 
-randomGcmToken :: MonadIO m => ClientId -> m PushToken
-randomGcmToken c = randomToken c GCM appName
+data TokenSpec = TokenSpec {tSize :: Int, tName :: AppName}
 
-randomToken :: MonadIO m => ClientId -> Transport -> AppName -> m PushToken
-randomToken c trans name = liftIO $ do
+defTS :: TokenSpec
+defTS = TokenSpec 32 appName
+
+randomGcmToken :: MonadIO m => ClientId -> TokenSpec -> m PushToken
+randomGcmToken c ts = randomToken c (tSize ts) GCM (tName ts)
+
+randomApnsToken :: MonadIO m => ClientId -> TokenSpec -> m PushToken
+randomApnsToken c ts = randomToken c (tSize ts) APNSSandbox (tName ts)
+
+randomToken :: MonadIO m => ClientId -> Int -> Transport -> AppName -> m PushToken
+randomToken c size trans name = liftIO $ do
     tok <- Token . T.decodeUtf8 <$> case trans of
-        GCM -> toByteString' <$> randomId
-        _   -> B16.encode    <$> randomBytes 32
+        GCM -> multipleUuids size randomId
```

Comment:
Instead of using the size of a uuid, how about just specifying the size in `Bytes`?

Reply:
GCM tokens and APNS tokens were different sizes in this test previously. I think it will still work with specifying the same size for both, but I wanted to keep it consistent with the previous behaviour.